pkg/ext-proc/backend/pod_reconciler.go

```diff
@@ -56,15 +55,17 @@ func (c *PodReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Complete(c)
 }
 
-func (c *PodReconciler) updateDatastore(k8sPod *corev1.Pod, inferencePool *v1alpha1.InferencePool) {
-	pod := Pod{
-		Name:    k8sPod.Name,
-		Address: k8sPod.Status.PodIP + ":" + strconv.Itoa(int(inferencePool.Spec.TargetPortNumber)),
-	}
-	if !k8sPod.DeletionTimestamp.IsZero() || !c.Datastore.LabelsMatch(k8sPod.ObjectMeta.Labels) || !podIsReady(k8sPod) {
-		c.Datastore.pods.Delete(pod)
```

Comment:
This is another bug introduced by using the backend.Pod as a key, which takes into account the address and the port. If any of those changes (e.g., the pod is deleted and recreated with the same name like statefulsets or the port on the inferencePool changes), we wouldn't be able to lookup and delete the previous version of the pod.